### DISCUSSION TITLE
How to debug flickering/vertical vibration?

### DISCUSSION THREAD

**Discussion by RomanTheLegend** [Roger-random/ESP_8_BIT_composite#39] [Q&A] [answered]
Good afternoon, could please anyone provide some guidance on how to debug and fix the problem with uncontrolled flickering and line position of the screen.

I'm porting the code that worked fine on TFT display of LilyGo T-GO to the composite-video driven device and I ran into the issue where the text is jumping left-right (actually, up-down because 270 rotation) without any apparent reason:

https://user-images.githubusercontent.com/20524417/193466746-59005bd8-36bd-44dc-8335-f6ccf7585fa1.mp4

(don't mind the color lines-this is just camera artifact)
The same is happening regardless of `copyAfterSwap` flag set

When running examples provided with the library, everything seem to be fine and text or pictures remain in place.

Some explanations:
Based on TFT-based code, I publish each text line separately:
https://github.com/RomanTheLegend/WingsuitGps/blob/composite_display/src/modes/MainMenuMode.cpp
and then do not push any new data to framebuffer unless its changed, so the main loop operates on the same buffers

That also forces me to call `waitForFrame`() in method `displayString` twice to make sure every line is displayed (otherwise the last line stays in buffer undisplayed)
https://github.com/RomanTheLegend/WingsuitGps/blob/composite_display/src/devices/CompositeDisplay.cpp

**Comment by Roger-random** [comment]
First suggestion is to narrow down the scope of the problem by testing different variables.

* Does this issue persist with different composite video display hardware?
* Does this issue persist with ESP_8_BIT? (https://github.com/rossumur/esp_8_bit)
* Does this issue persist with different ESP32 composite video libraries? (https://github.com/Roger-random/ESP_8_BIT_composite#alternatives)

**Comment by ghost** [comment]
Also, try different models of board in the Arduino IDE.  I had a problem like this once and it turned out to be the model of esp32 selected.

**Comment by RomanTheLegend** [comment, marked as answer]
Apologies for the long reply, I had to test out many things to get to the root cause. Turned out the problem is with Platformio's compiler which caused this flickering, and that was the reason why your basic Arduino examples on the other hand were working just fine

For those who wondering, add these lines to your `platformio.ini` file:

```
platform = espressif32@5.2.0
platform_packages =
   framework-arduinoespressif32 @ https://github.com/espressif/arduino-esp32.git
```

Update: another flag might cause waving distortion effects to the output, so be aware of setting `build_type = debug`

**Reply by Roger-random** [reply]
Thank you for your investigation work and sharing your findings!
